### PR TITLE
TF-TRT test FusedBatchNorm op converter

### PR DESCRIPTION
**Pull request by tfeher**
Add test for the FusedBatchNorm op converter in implicit batch, explicit batch and dynamic shape mode. Tagging @bixia1 for review.

### CODE REVIEW

**Comment by gbaned** [comment]
@bixia1 Can you please review this PR ? Thanks!

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc [hunk 1]
```diff
@@ -2011,6 +2011,127 @@ TEST_F(OpConverterTest, ConvertConst) {
...
+               "Channel dimension must be static, at my_batchnorm"),
+           true});
+    }
+    for (auto p : test_param) {
+      Reset();
+      NodeDef node_def =
+          get_node_def(tf_type, p.data_format, p.is_training, p.epsilon);
+      for (int i = 0; i < node_input.size(); i++) {
+        if (i == 0 || i == p.tensor_input_idx) {
+          // Add input i as a tensor. The converter expects that only the first
```
Comment (bixia1) [inline tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc:2102]: I think this comment should explain the following statement: that is the reason why we can't add the tensor is i !=0 for implicit batch mode. And the reason is when we add input 0, we already set up the batch size of the converter to 2. All the rest tensor_input_idx corresponding to batch size 3 in test_param and node_input. For this reason, I would change the comment to:
// In node_input, all tensors have batch size == 3 except for the first tensor. Therefore, adding tensors with tensor_input_index !=1 as input should cause mismatching batch size for implicit batch mode.

**Comment by gbaned** [comment]
@tfeher Can you please check @bixia1's comments and keep us posted. Thanks!

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc (hunk 1 above)
Comment (tfeher) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes_test.cc]: I have added a longer version of this as a comment.

**Review by tfeher — COMMENTED** [review]
Thanks @bixia1 for the review, I have addressed the issues.

**Review by bixia1 — APPROVED** [review]

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc [hunk 2]
```diff
@@ -4942,7 +4942,10 @@ Status ConvertFusedBatchNorm(OpConverterParams* params) {
                                  node_def.name());
   }
   nvinfer1::ITensor* tensor = inputs.at(0).tensor();
-
+  if (!params->use_implicit_batch && tensor->getDimensions().d[1] == -1) {
```
[the remark this answers is not in the document]
Comment (bixia1) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc:4945]: Thanks for clarifying this. I would suggest a comment like this:
    // This check is to make sure that channel dimension is known during
    // conversion.
    //
    // We check this only in explicit batch mode and reject an op with unknown
    // channel dimension during segmentation. In implicit batch mode we have
    // known shapes during conversion even though the shapes may not be known
    // during segmentation (see the actual argument for input_shapes when
    // ConvertGraphDefToEngine is called from TRTEngineOp::BuildEngine).

**Review by bixia1 — CHANGES_REQUESTED** [review]
Sorry. I was in the process of merging this and realize that I need to make non-trivial changes to the comment for ConvertFusedBatchNorm. Need to get your agreement on this.

**Comment by gbaned** [comment]
@tfeher Can you please check @bixia1's comments and keep us posted ? Thanks!

File: tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc (hunk 2 above)
[the remark this answers is not in the document]
Comment (tfeher) [reply tensorflow/compiler/tf2tensorrt/convert/convert_nodes.cc:4945]: I have updated the comment as you have suggested.

**Comment by tfeher** [comment]
Thanks @bixia1, I have made the requested changes.

**Review by bixia1 — APPROVED** [review]

**Pull request merged**